pycargoebuild/cargo.py

```diff
@@ -56,18 +57,62 @@ def crate_entry(self) -> str:
         return f"{self.name}@{self.version}"
 
 
+class GitHost(enum.Enum):
+    GITHUB = enum.auto()
+    GITLAB = enum.auto()
+    GITLAB_SELFHOSTED = enum.auto()
+
+
 @dataclasses.dataclass(frozen=True)
 class GitCrate(Crate):
     repository: str
     commit: str
 
+    def __post_init__(self) -> None:
+        self.repo_host  # check for supported git hosts
+
+    @property
```

Comment:
```suggestion
    @cached_property
```

perhaps? My idea was to store it at init, but caching would also work.